src/JuliaSyntaxHighlighting.jl

```diff
@@ -59,96 +69,232 @@ const HIGHLIGHT_FACES = [
-            else
-                depth0 = getfield(depthcounters, ptype)[]
-                getfield(depthcounters, ptype)[] += depthchange
-                depth0
+        end
+    elseif nkind == K"quote" && length(node.args) == 2 &&
+        kind(node.args[1]) == K":" && kind(node.args[2]) == K"Identifier"
+        :julia_symbol
+    elseif nkind == K"Comment"; :julia_comment
+    elseif nkind == K"String"; :julia_string
+    elseif JuliaSyntax.is_string_delim(node); :julia_string_delim
+    elseif nkind == K"CmdString"; :julia_cmdstring
+    elseif nkind == K"`" || nkind == K"```"; :julia_cmdstring
+    elseif nkind == K"Char"
+        kind(lnode) == K"'" && !isempty(highlights) &&
+            (highlights[end] = (highlights[end][1], :face => :julia_char_delim))
+        :julia_char
+    elseif nkind == K"'" && kind(lnode) == K"Char"; :julia_char_delim
+    elseif nkind == K"true" || nkind == K"false"; :julia_bool
+    elseif JuliaSyntax.is_number(nkind); :julia_number
+    elseif JuliaSyntax.is_prec_assignment(nkind) && JuliaSyntax.is_trivia(node);
+        if nkind == K"="
+            ifelse(ppkind == K"for", :julia_keyword, :julia_assignment)
+        else # updating for <op>=
+            push!(highlights, (firstindex(content)+offset:node.span+offset-1, :face => :julia_operator))
+            push!(highlights, (node.span+offset:node.span+offset, :face => :julia_assignment))
+            nothing
+        end
+    elseif nkind == K";" && pkind == K"parameters" && pnode == lnode
+        :julia_assignment
+    elseif (JuliaSyntax.is_keyword(nkind) ||nkind == K"->" ) && JuliaSyntax.is_trivia(node)
+        :julia_keyword
+    elseif nkind == K"where"
+        if JuliaSyntax.is_trivia(node)
+            :julia_keyword
+        else
+            literal_where = findfirst(
+                c ->kind(c) == K"where" && JuliaSyntax.is_trivia(c),
+                node.args)
+            if !isnothing(literal_where)
+                shift = sum(c ->Int(c.span), node.args[1:literal_where])
+                region = first(region)+shift:last(region)
+                :julia_type
             end
-            if pdepth <= 0 && UNMATCHED_DELIMITERS_ENABLED[]
-                :julia_unpaired_parenthetical
-            elseif !RAINBOW_DELIMITERS_ENABLED[]
-                :julia_parenthetical
-            else
-                displaydepth = mod1(pdepth, MAX_PAREN_HIGHLIGHT_DEPTH)
-                Symbol("julia_rainbow_$(ptype)_$(displaydepth)")
+        end
+    elseif nkind == K"in"
+        ifelse(ppkind == K"for", :julia_keyword, :julia_comparator)
+    elseif nkind == K"isa"; :julia_builtin
+    elseif nkind in (K"&&", K"||", K"<:", K"===") && JuliaSyntax.is_trivia(node)
+        :julia_builtin
+    elseif JuliaSyntax.is_prec_comparison(nkind) && JuliaSyntax.is_trivia(node);
+        :julia_comparator
+    elseif isplainoperator(node); :julia_operator
+    elseif nkind == K"..." && JuliaSyntax.is_trivia(node); :julia_operator
```

Comment:
This whole thing might be cleaner if you split the big `elseif` thing into two sections based on `haschildren()`?

Because, for example, `K"for"` means two different things, depending on whether you're looking at a token (`!haschildren()` - ie, the literal `for` trivia token in the source) vs an interior node of the tree representing a for loop, with the comparision and body as children.

Reply:
Hmmm, that's a good point. I might leave this for a future refactor (I'm sure there will be a few updates from this initial rewrite) though.